packages/jam/transition/externalities/fetch-externalities.ts

```diff
@@ -210,62 +283,109 @@ export class FetchExternalities implements IFetchExternalities {
-    if (this.fetchData.context !== FetchContext.Accumulate) {
-      return null;
-    }
+    if (this.fetchData.context === FetchContext.LegacyAccumulate) {
+      const { operands } = this.fetchData;
 
-    const { operands } = this.fetchData;
+      if (operandIndex >= 2n ** 32n) {
+        return null;
+      }
 
-    if (operandIndex >= 2n ** 32n) {
-      return null;
-    }
+      const operand = operands[Number(operandIndex)];
 
-    const operand = operands[Number(operandIndex)];
+      if (operand === undefined) {
+        return null;
+      }
 
-    if (operand === undefined) {
-      return null;
+      return Encoder.encodeObject(Operand.Codec, operand, this.chainSpec);
     }
 
-    return Encoder.encodeObject(Operand.Codec, operand, this.chainSpec);
+    return null;
   }
 
   allTransfers(): BytesBlob | null {
-    if (this.fetchData.context !== FetchContext.OnTransfer) {
-      return null;
-    }
+    if (this.fetchData.context === FetchContext.LegacyOnTransfer) {
+      const { transfers } = this.fetchData;
 
-    const { transfers } = this.fetchData;
+      return Encoder.encodeObject(codec.sequenceVarLen(PendingTransfer.Codec), transfers, this.chainSpec);
+    }
 
-    return Encoder.encodeObject(codec.sequenceVarLen(PendingTransfer.Codec), transfers, this.chainSpec);
+    return null;
   }
 
   oneTransfer(transferIndex: U64): BytesBlob | null {
-    if (this.fetchData.context !== FetchContext.OnTransfer) {
-      return null;
+    if (this.fetchData.context === FetchContext.LegacyOnTransfer) {
+      const { transfers } = this.fetchData;
+
+      if (transferIndex >= 2n ** 32n) {
+        return null;
+      }
+
+      const transfer = transfers[Number(transferIndex)];
+
+      if (transfer === undefined) {
+        return null;
+      }
+
+      return Encoder.encodeObject(PendingTransfer.Codec, transfer, this.chainSpec);
     }
 
-    const { transfers } = this.fetchData;
+    return null;
+  }
 
-    if (transferIndex >= 2n ** 32n) {
-      return null;
+  allTransfersAndOperands(): BytesBlob | null {
+    if (this.fetchData.context === FetchContext.Accumulate) {
+      const { transfers, operands } = this.fetchData;
+      const transfersAndOperands: TransferOrOperand[] = transfers
+        .map((transfer): TransferOrOperand => ({ kind: TransferOperandKind.TRANSFER, value: transfer }))
+        .concat(operands.map((operand): TransferOrOperand => ({ kind: TransferOperandKind.OPERAND, value: operand })));
+
+      return Encoder.encodeObject(TRANSFERS_AND_OPERANDS, transfersAndOperands, this.chainSpec);
     }
 
-    const transfer = transfers[Number(transferIndex)];
+    return null;
+  }
+
+  oneTransferOrOperand(index: U64): BytesBlob | null {
+    if (this.fetchData.context === FetchContext.Accumulate) {
+      const { operands, transfers } = this.fetchData;
 
-    if (transfer === undefined) {
-      return null;
+      if (index >= operands.length + transfers.length || index >= 2n ** 32n) {
```

Comment:
```suggestion
      if (index >= operands.length + transfers.length) {
```
I think that should be enough?